receiver/faroreceiver/factory.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faroreceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
+)
+
+const (
+	typeStr   = "faro"
+	stability = component.StabilityLevelDevelopment
+
+	defaultHTTPEndpoint = "0.0.0.0:8886"
+)
+
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithLogs(createLogReceiver, stability),
+		receiver.WithTraces(createTraceReceiver, stability),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		HTTP: &confighttp.HTTPServerSettings{
```

Comment:
Is there a reason to have the http server settings be nested under HTTP? Most just extend it.